.ci/update-github-page.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+set -e
+
+source ./.ci/util.sh
+
+checkForVariable "GITHUB_TOKEN"
+
+checkout_from https://github.com/checkstyle/contribution
+
+cd .ci-temp/contribution/releasenotes-builder
+mvn -e --no-transfer-progress clean compile package
+cd ../../../
+
+if [ -d .ci-temp/checkstyle ]; then
+  cd .ci-temp/checkstyle/
+  git reset --hard origin/master
+  git pull origin master
+  git fetch --tags
+  cd ../../
+else
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/checkstyle
+  cd ../
+fi
+
+cd .ci-temp/checkstyle
+
+curl \
+ https://api.github.com/repos/checkstyle/checkstyle/releases \
+ -H "Authorization: token $GITHUB_TOKEN" \
+ -o /var/tmp/cs-releases.json
+
+TARGET_RELEASE_NUM=10.1
+TARGET_RELEASE_INDEX=$(cat /var/tmp/cs-releases.json | \
+    jq "[.[].tag_name] | to_entries | .[] | \
+    select(.value==\"checkstyle-$TARGET_RELEASE_NUM\") | .key")
+echo TARGET_RELEASE_INDEX="$TARGET_RELEASE_INDEX"
+
+PREVIOUS_RELEASE_INDEX=$(($TARGET_RELEASE_INDEX+1))
+echo PREVIOUS_RELEASE_INDEX="$PREVIOUS_RELEASE_INDEX"
+
+END_REF=$(cat /var/tmp/cs-releases.json | jq -r ".[$TARGET_RELEASE_INDEX].tag_name")
+START_REF=$(cat /var/tmp/cs-releases.json | jq -r ".[$PREVIOUS_RELEASE_INDEX].tag_name")
+
+echo START_REF="$START_REF"
+echo END_REF="$END_REF"
+
+cd ../
+
+BUILDER_RESOURCE_DIR="contribution/releasenotes-builder/src/main/resources/com/github/checkstyle"
+
+java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.jar \
+     -localRepoPath checkstyle \
+     -remoteRepoPath checkstyle/checkstyle \
+     -startRef "$START_REF" \
+     -endRef "$END_REF" \
+     -releaseNumber "$CS_RELEASE_VERSION" \
```

Comment:
this varible isnot defined .... so it is a reason of missed version print in output

Reply:
Please unset all variables that you use in script. To test clearly.

Reply:
Please NOTE: Please don't use `&&` for testing

Testing:
```
rahul@rk7:~/Desktop/opensource/checkstyle$ READ_ONLY_TOKEN=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 GITHUB_TOKEN=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  ./.ci/update-github-page.sh 10.1
Cloning into 'contribution'...
remote: Enumerating objects: 7956, done.
remote: Counting objects: 100% (1230/1230), done.
remote: Compressing objects: 100% (539/539), done.
remote: Total 7956 (delta 628), reused 1098 (delta 561), pack-reused 6726
Receiving objects: 100% (7956/7956), 2.86 MiB | 1.93 MiB/s, done.
Resolving deltas: 100% (4086/4086), done.
...
...
...
[INFO] Attaching shaded artifact.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.413 s
[INFO] Finished at: 2022-08-21T17:57:56+05:30
[INFO] ------------------------------------------------------------------------
Cloning into 'checkstyle'...
...
...
....
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  115k  100  115k    0     0  30853      0  0:00:03  0:00:03 --:--:-- 30851
TARGET_RELEASE_INDEX=4
PREVIOUS_RELEASE_INDEX=5
START_REF=checkstyle-10.0
END_REF=checkstyle-10.1

[WARN] Issue #6207 "Expand XPath IT Regression Testing" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/6207
[WARN] Issue #10924 "All checks that use `AbstractCheck#getLine()` should check code points instead of characters for spacing" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/10924
[WARN] Issue #11446 "Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11446
[WARN] Issue #11214 "Specify violation messages in input files." is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11214
[WARN] Issue #4845 "Test inputs should be completely standalone." is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/4845
[WARN] Issue #11166 "Fix existing cases of `getFileContents()` usage" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11166

Execution succeeded!
==============================================
GITHUB PAGE:
==============================================
Checkstyle 10.1 - https://checkstyle.org/releasenotes.html#Release_10.1

Breaking backward compatibility:

  #9146 - upgrade minimal required jdk version to 11
  #10907 - LineLengthCheck ignores length of import and package statements.

New:

  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens

Bug fixes:

  #10924 - All checks that use `AbstractCheck#getLine()` should check code points instead of characters for spacing
  #11213 - False positive: SummaryJavadocCheck
  #10776 - Inline @return tag does not pass SummaryJavadocCheck
  #11357 - UnnecessaryParentheses reports incorrect violation message for return value in lambdas
  #11030 - Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.

<details>
<summary>Other Changes:</summary>
<br/>
  Expand XPath IT Regression Testing <br/>
<br/>
  Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files <br/>
<br/>
  Conver test for SuppressionXpathSingleFilterTest to use inlined config in Input files <br/>
<br/>
  Conver test for SuppressionXpathFilterTest to use inlined config in Input files <br/>
<br/>
  Specify violation messages in input files. <br/>
<br/>
  Update doc for MultipleStringLiterals <br/>
<br/>
  Move methods common in UnusedLocalVariableCheck and FinalClassCheck to util <br/>
<br/>
  Remove suppressions for original verify method or explain in comment <br/>
<br/>
  Migrate to Inline Config Parser for filters <br/>
<br/>
  Test inputs should be completely standalone. <br/>
<br/>
  spelling: Fix double word in TreeWalker <br/>
<br/>
  Checkstyle should force logging of AST in checks that require it <br/>
<br/>
  refactor check for override annotation - add util method <br/>
<br/>
  Fix existing cases of `getFileContents()` usage <br/>
<br/>
  JavadocMetadataScraper fails silently if it can't write the XML files <br/>
<br/>
  AvoidStarImport: update documentation with example for mix of properties  <br/>
<br/>
  doc: mention that excludes can be class names <br/>
</details>
Updating content to be be json value friendly
Checkstyle 10.1 - https://checkstyle.org/releasenotes.html#Release_10.1\n\nBreaking backward compatibility:\n\n  #9146 - upgrade minimal required jdk version to 11\n  #10907 - LineLengthCheck ignores length of import and package statements.\n\nNew:\n\n  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens\n\nBug fixes:\n\n  #10924 - All checks that use 'AbstractCheck#getLine()' should check code points instead of characters for spacing\n  #11213 - False positive: SummaryJavadocCheck\n  #10776 - Inline @return tag does not pass SummaryJavadocCheck\n  #11357 - UnnecessaryParentheses reports incorrect violation message for return value in lambdas\n  #11030 - Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.\n\n<details>\n<summary>Other Changes:</summary>\n<br/>\n  Expand XPath IT Regression Testing <br/>\n<br/>\n  Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathSingleFilterTest to use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathFilterTest to use inlined config in Input files <br/>\n<br/>\n  Specify violation messages in input files. <br/>\n<br/>\n  Update doc for MultipleStringLiterals <br/>\n<br/>\n  Move methods common in UnusedLocalVariableCheck and FinalClassCheck to util <br/>\n<br/>\n  Remove suppressions for original verify method or explain in comment <br/>\n<br/>\n  Migrate to Inline Config Parser for filters <br/>\n<br/>\n  Test inputs should be completely standalone. <br/>\n<br/>\n  spelling: Fix double word in TreeWalker <br/>\n<br/>\n  Checkstyle should force logging of AST in checks that require it <br/>\n<br/>\n  refactor check for override annotation - add util method <br/>\n<br/>\n  Fix existing cases of 'getFileContents()' usage <br/>\n<br/>\n  JavadocMetadataScraper fails silently if it can't write the XML files <br/>\n<br/>\n  AvoidStarImport: update documentation with example for mix of properties  <br/>\n<br/>\n  doc: mention that excludes can be class names <br/>\n</details>\n
RELEASE_ID=73185822
{
"tag_name": "checkstyle-10.1",
"body": "Checkstyle 10.1 - https://checkstyle.org/releasenotes.html#Release_10.1\n\nBreaking backward compatibility:\n\n  #9146 - upgrade minimal required jdk version to 11\n  #10907 - LineLengthCheck ignores length of import and package statements.\n\nNew:\n\n  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens\n\nBug fixes:\n\n  #10924 - All checks that use 'AbstractCheck#getLine()' should check code points instead of characters for spacing\n  #11213 - False positive: SummaryJavadocCheck\n  #10776 - Inline @return tag does not pass SummaryJavadocCheck\n  #11357 - UnnecessaryParentheses reports incorrect violation message for return value in lambdas\n  #11030 - Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.\n\n<details>\n<summary>Other Changes:</summary>\n<br/>\n  Expand XPath IT Regression Testing <br/>\n<br/>\n  Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathSingleFilterTest to use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathFilterTest to use inlined config in Input files <br/>\n<br/>\n  Specify violation messages in input files. <br/>\n<br/>\n  Update doc for MultipleStringLiterals <br/>\n<br/>\n  Move methods common in UnusedLocalVariableCheck and FinalClassCheck to util <br/>\n<br/>\n  Remove suppressions for original verify method or explain in comment <br/>\n<br/>\n  Migrate to Inline Config Parser for filters <br/>\n<br/>\n  Test inputs should be completely standalone. <br/>\n<br/>\n  spelling: Fix double word in TreeWalker <br/>\n<br/>\n  Checkstyle should force logging of AST in checks that require it <br/>\n<br/>\n  refactor check for override annotation - add util method <br/>\n<br/>\n  Fix existing cases of 'getFileContents()' usage <br/>\n<br/>\n  JavadocMetadataScraper fails silently if it can't write the XML files <br/>\n<br/>\n  AvoidStarImport: update documentation with example for mix of properties  <br/>\n<br/>\n  doc: mention that excludes can be class names <br/>\n</details>\n"
}
Updating Github tag page
{
  "url": "https://api.github.com/repos/Rahulkhinchi03/checkstyle/releases/73185822",
  "assets_url": "https://api.github.com/repos/Rahulkhinchi03/checkstyle/releases/73185822/assets",
  "upload_url": "https://uploads.github.com/repos/Rahulkhinchi03/checkstyle/releases/73185822/assets{?name,label}",
  "html_url": "https://github.com/Rahulkhinchi03/checkstyle/releases/tag/checkstyle-10.1",
  "id": 73185822,
  "author": {
    "login": "Rahulkhinchi03",
    "id": 71710042,
    "node_id": "MDQ6VXNlcjcxNzEwMDQy",
    "avatar_url": "https://avatars.githubusercontent.com/u/71710042?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/Rahulkhinchi03",
    "html_url": "https://github.com/Rahulkhinchi03",
    "followers_url": "https://api.github.com/users/Rahulkhinchi03/followers",
    "following_url": "https://api.github.com/users/Rahulkhinchi03/following{/other_user}",
    "gists_url": "https://api.github.com/users/Rahulkhinchi03/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/Rahulkhinchi03/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/Rahulkhinchi03/subscriptions",
    "organizations_url": "https://api.github.com/users/Rahulkhinchi03/orgs",
    "repos_url": "https://api.github.com/users/Rahulkhinchi03/repos",
    "events_url": "https://api.github.com/users/Rahulkhinchi03/events{/privacy}",
    "received_events_url": "https://api.github.com/users/Rahulkhinchi03/received_events",
    "type": "User",
    "site_admin": false
  },
  "node_id": "RE_kwDOHuKuTs4EXLoe",
  "tag_name": "checkstyle-10.1",
  "target_commitish": "master",
  "name": "checkstyle-1.2",
  "draft": false,
  "prerelease": false,
  "created_at": "2022-03-27T14:51:31Z",
  "published_at": "2022-07-29T16:24:08Z",
  "assets": [

  ],
  "tarball_url": "https://api.github.com/repos/Rahulkhinchi03/checkstyle/tarball/checkstyle-10.1",
  "zipball_url": "https://api.github.com/repos/Rahulkhinchi03/checkstyle/zipball/checkstyle-10.1",
  "body": "Checkstyle 10.1 - https://checkstyle.org/releasenotes.html#Release_10.1\n\nBreaking backward compatibility:\n\n  #9146 - upgrade minimal required jdk version to 11\n  #10907 - LineLengthCheck ignores length of import and package statements.\n\nNew:\n\n  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens\n\nBug fixes:\n\n  #10924 - All checks that use 'AbstractCheck#getLine()' should check code points instead of characters for spacing\n  #11213 - False positive: SummaryJavadocCheck\n  #10776 - Inline @return tag does not pass SummaryJavadocCheck\n  #11357 - UnnecessaryParentheses reports incorrect violation message for return value in lambdas\n  #11030 - Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.\n\n<details>\n<summary>Other Changes:</summary>\n<br/>\n  Expand XPath IT Regression Testing <br/>\n<br/>\n  Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathSingleFilterTest to use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathFilterTest to use inlined config in Input files <br/>\n<br/>\n  Specify violation messages in input files. <br/>\n<br/>\n  Update doc for MultipleStringLiterals <br/>\n<br/>\n  Move methods common in UnusedLocalVariableCheck and FinalClassCheck to util <br/>\n<br/>\n  Remove suppressions for original verify method or explain in comment <br/>\n<br/>\n  Migrate to Inline Config Parser for filters <br/>\n<br/>\n  Test inputs should be completely standalone. <br/>\n<br/>\n  spelling: Fix double word in TreeWalker <br/>\n<br/>\n  Checkstyle should force logging of AST in checks that require it <br/>\n<br/>\n  refactor check for override annotation - add util method <br/>\n<br/>\n  Fix existing cases of 'getFileContents()' usage <br/>\n<br/>\n  JavadocMetadataScraper fails silently if it can't write the XML files <br/>\n<br/>\n  AvoidStarImport: update documentation with example for mix of properties  <br/>\n<br/>\n  doc: mention that excludes can be class names <br/>\n</details>\n",
  "mentions_count": 1
}
rahul@rk7:~/Desktop/opensource/checkstyle$ 
```